lib/instrumentation/express-utils.js

```diff
@@ -28,7 +34,13 @@ function getStackPath (req) {
 
 // This function is also able to extract the path from a Restify request as
 // it's storing the route name on req.route.path as well
-function getPathFromRequest (req, useBase) {
+function getPathFromRequest (req, useBase, usePathAsTransactionName) {
+  // Enable usePathAsTransactionName config
+  if (usePathAsTransactionName) {
+    const parsed = parseUrl(req)
+    return parsed && parsed.pathname
+  }
```

Comment:
Do we want this at the start rather than at the end as a fallback when no other viable route name is determined?

Reply:
That was my idea yes. I did consider what you are suggesting, but the config option is called `usePathAsTransactionName`, so I think users would expect that if they set it to `true` that it would overrule everything.

Reply:
🤔 on the other hand, now that I re-read [the docs](https://www.elastic.co/guide/en/apm/agent/java/current/config-http.html#config-use-path-as-transaction-name) for the same config option in Java, it does seem like they use it as a fallback. Is that correctly understood @felixbarny?

Reply:
docs said `If your URLs contain path parameters like /user/$userId, you should be very careful when enabling this flag, as it can lead to an explosion of transaction groups. `  
But if its a fallback when no other viable route name is determined, all ` /user/$userId` will be in one group, its not what the docs say

Reply:
In the Java agent, this option is only relevant if no other route could be determined. The default transaction name would then be just `GET` if this option is disabled. If it's enabled, it would then use the URL path, for example, `/users/1`, `/users/2` etc. That's why there is also [`url_groups`](https://www.elastic.co/guide/en/apm/agent/java/current/config-http.html#config-url-groups). If you define the pattern `/users/*`, the URLs will be grouped so that there won't be an explosion of transaction groups.

Reply:
@yuxizhe no worries. You implemented it as we had agreed. And it's not wrong per se. But we just realized that the Java agent actually does it slightly different. So now we're just discussing if we should update this code (and docs) to do what the Java agent does.

Reply:
I put it at the end as a fallback and update docs. I think we need url_groups config too.

Reply:
Thanks for updating. I talked to @felixbarny today and we discussed that we at some point should probably replace this config option for one that can be set to both works as a fallback and as an overwrite - both use-cases are valid. But for now, your latest change is fine. At least this means that we align with the Java agent 😃 

At some point, we'll probably implement `urlGroups` as well, but I don't see that being a showstopper for this PR 👍